modules/swagger-codegen/src/main/java/com/wordnik/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -71,9 +58,66 @@ public JavaClientCodegen() {
       );
     instantiationTypes.put("array", "ArrayList");
     instantiationTypes.put("map", "HashMap");
+    
+    cliOptions.add(new CliOption("invokerPackage", "root package for generated code"));
+    cliOptions.add(new CliOption("groupId", "groupId in generated pom.xml"));
+    cliOptions.add(new CliOption("artifactId", "artifactId in generated pom.xml"));
+    cliOptions.add(new CliOption("artifactVersion", "artifact version in generated pom.xml"));
+    cliOptions.add(new CliOption("sourceFolder", "source folder for generated code"));
   }
 
   @Override
+  public void processOpts() {
+    super.processOpts();
+    
+    if(additionalProperties.containsKey("invokerPackage")) {
+      this.setInvokerPackage((String)additionalProperties.get("invokerPackage"));
+    }
+    else{
+      //not set, use default to be passed to template
+      additionalProperties.put("invokerPackage", invokerPackage);
+    }
```

Comment:
@xhh I did remove the template generated comments, thanks.

RE: reading and storing config options in `DefaultCodegen`
    Right now there are two config options supported by all of the languages, `modelPackage` and `apiPackage`, those are being read and set in  the `processOpts` method of `DefaultCodegen`, all the other config options that are language specific should be managed by that lang specific config itself. If in the the future there is another config option that is supported by all the languages then it should be managed it DefaultCodegen.

RE: validation
    No validation is necessary on the config options. Even if model package does not contain spaces it still can be out of sync with apiPackage and it will not compile. Using these custom config options assumes some level of competency.